egs2/TEMPLATE/sds1/app.py

```diff
@@ -0,0 +1,797 @@
+import argparse
+import os
+import shutil
+import time
+
+import gradio as gr
+import nltk
+import numpy as np
+import torch
+from huggingface_hub import HfApi
+from pyscripts.utils.dialog_eval.ASR_WER import handle_espnet_ASR_WER
+from pyscripts.utils.dialog_eval.human_feedback import (
+    natural_vote1_last_response,
+    natural_vote2_last_response,
+    natural_vote3_last_response,
+    natural_vote4_last_response,
+    relevant_vote1_last_response,
+    relevant_vote2_last_response,
+    relevant_vote3_last_response,
+    relevant_vote4_last_response,
+)
+from pyscripts.utils.dialog_eval.LLM_Metrics import (
+    DialoGPT_perplexity,
+    bert_score,
+    perplexity,
+    vert,
+)
+from pyscripts.utils.dialog_eval.TTS_intelligibility import (
+    handle_espnet_TTS_intelligibility,
+)
+from pyscripts.utils.dialog_eval.TTS_speech_quality import TTS_psuedomos
+
+from espnet2.sds.espnet_model import ESPnetSDSModelInterface
+
+access_token = os.environ.get("HF_TOKEN")
+ASR_name = None
+LLM_name = None
+TTS_name = None
+ASR_options = []
+LLM_options = []
+TTS_options = []
+Eval_options = []
+upload_to_hub = None
+dialogue_model = None
+
+
+def read_args():
+    global access_token
+    global ASR_name
+    global LLM_name
+    global TTS_name
+    global ASR_options
+    global LLM_options
+    global TTS_options
+    global Eval_options
+    global upload_to_hub
+    global dialogue_model
+    parser = argparse.ArgumentParser(
+        description="Run the app with HF_TOKEN as a command-line argument."
+    )
+    parser.add_argument(
+        "--HF_TOKEN", required=True, help="Provide the Hugging Face token."
+    )
+    parser.add_argument(
+        "--asr_options",
+        required=True,
+        help="Provide the possible ASR options available to user.",
+    )
+    parser.add_argument(
+        "--llm_options",
+        required=True,
+        help="Provide the possible LLM options available to user.",
+    )
+    parser.add_argument(
+        "--tts_options",
+        required=True,
+        help="Provide the possible TTS options available to user.",
+    )
+    parser.add_argument(
+        "--eval_options",
+        required=True,
+        help="Provide the possible automatic evaluation metrics available to user.",
+    )
+    parser.add_argument(
+        "--default_asr_model",
+        required=False,
+        default="pyf98/owsm_ctc_v3.1_1B",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_llm_model",
+        required=False,
+        default="meta-llama/Llama-3.2-1B-Instruct",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--default_tts_model",
+        required=False,
+        default="kan-bayashi/ljspeech_vits",
+        help="Provide the default ASR model.",
+    )
+    parser.add_argument(
+        "--upload_to_hub",
+        required=False,
+        default=None,
+        help="Hugging Face dataset to upload user data",
```

Comment:
Could you clarify more about this flag? I couldn't understand the role of this flag...